### PR TITLE
Entrance of auto schedule

### PR DESCRIPTION
**Pull request by CtfGo**
1. add `AutoTuner` class as the main interface for user to call auto-schedule to tune graph or schedule.
2. adjust the role of `TaskScheduler`, it will only return the next task id to tune and not responsible for scheduling task to tune.
3. add associated simple unit tests.

### CODE REVIEW

File: cinn/auto_schedule/auto_tuner.cc [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+namespace cinn {
+namespace auto_schedule {
+
+AutoTuner::AutoTuner(const common::Target& target, hlir::framework::Graph* graph) : target_(target), graph_(graph) {}
+
+void AutoTuner::Initialize(const Config& config, hlir::framework::GraphCompiler* graph_compiler) {
+  // create tasks
+  TaskCreator task_creator;
+  tasks_ = task_creator.CreateTuneTaskOpLevel(graph_);
+  CHECK_GT(tasks_.size(), 0) << "Create tasks failed";
```
Comment (zhhsplendid) [inline cinn/auto_schedule/auto_tuner.cc:31]: "Create tasks failed" -> "Create autotune tasks failed"

File: cinn/auto_schedule/auto_tuner.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+#include "cinn/auto_schedule/task_scheduler/task_scheduler.h"
+#include "cinn/auto_schedule/tuning.h"
+#include "cinn/common/target.h"
+#include "cinn/hlir/framework/graph.h"
+#include "cinn/hlir/framework/graph_compiler.h"
+
+namespace cinn {
+namespace auto_schedule {
+
+// This class is entrance of auto-tune, user can use it
```
Comment (zhhsplendid) [inline cinn/auto_schedule/auto_tuner.h:32]: "user" -> "users"

File: cinn/auto_schedule/auto_tuner.h [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+// to tune graph (not supported yet) and search a schedule
```
Comment (zhhsplendid) [inline cinn/auto_schedule/auto_tuner.h:33]: Not search a schedule. Search a series of schedules.

File: cinn/auto_schedule/auto_tuner.h [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+// that maybe more likely to obtain better performance.
+// Internally, it create necessary components and use them to finish tuning.
```
Comment (zhhsplendid) [inline cinn/auto_schedule/auto_tuner.h:35]: "it create" -> "it creates"

File: cinn/auto_schedule/tuning.h [hunk 5]
```diff
@@ -14,16 +14,40 @@
...
 
+// Result of the tuning process
+struct TuningResult {
+  // Result of graph tuning
+  struct TunedGraph {
+    std::vector<std::vector<hlir::framework::Node*>> groups;
+  };
+
+  // Result of schedule tuning in CINN IR
+  struct OptimizedSchedule {
```
Comment (zhhsplendid) [inline cinn/auto_schedule/tuning.h:44]: This name is not good. Schedule is same "change process". So I thought we could use "OptimizedSchedule" to store some searched schedule parameters (not ready yet). Here, should we call it OptimizedLoweredFunc?

**Review by zhhsplendid — APPROVED** [review]
Above are naming or grammar comment, you can merge PR first and then change it in the next PR

File: cinn/auto_schedule/auto_tuner.cc (hunk 1 above)
Comment (CtfGo) [reply cinn/auto_schedule/auto_tuner.cc:31]: i will change it in next PR

File: cinn/auto_schedule/auto_tuner.h (hunk 4 above)
Comment (CtfGo) [reply cinn/auto_schedule/auto_tuner.h:35]: i will change it in next PR

File: cinn/auto_schedule/auto_tuner.h (hunk 3 above)
Comment (CtfGo) [reply cinn/auto_schedule/auto_tuner.h:33]: i will change it in next PR

File: cinn/auto_schedule/auto_tuner.h (hunk 2 above)
Comment (CtfGo) [reply cinn/auto_schedule/auto_tuner.h:32]: i will change it in next PR

File: cinn/auto_schedule/tuning.h (hunk 5 above)
Comment (CtfGo) [reply cinn/auto_schedule/tuning.h:44]: i will change it in next PR

**Pull request merged**